sdk/storage/inc/datalake/datalake_options.hpp

```diff
@@ -0,0 +1,1053 @@
+     *        otherwise, the uncommitted data is deleted after the flush operation.  The
+     *        default is false.  Data at offsets less than the specified position are
+     *        written to the file when flush succeeds, but this optional parameter allows
+     *        data after the flush position to be retained for a future flush operation.
+     */
+    bool RetainUncommittedData = bool();
+
+    /**
+     * @brief Azure Storage Events allow applications to receive notifications when files
+     *        change. When Azure Storage Events are enabled, a file changed event is raised.
+     *        This event has a property indicating whether this is the final change to distinguish
+     *        the difference between an intermediate flush to a file stream and the final close of
+     *        a file stream. The close query parameter is valid only when the action is "flush"
+     *        and change notifications are enabled. If the value of close is "true" and the
+     *        flush operation completes successfully, the service raises a file change notification
+     *        with a property indicating that this is the final update (the file stream has been
+     *        closed). If "false" a change notification is raised indicating the file has changed.
+     *        The default is false. This query parameter is set to true by the Hadoop ABFS driver to
+     *        indicate that the file stream has been closed."
+     */
+    bool Close = bool();
+
+    /**
+     * @brief The service stores this value and includes it in the "Content-Md5" response header for
+     *        "Read & Get Properties" operations. If this property is not specified on the request,
+     *        then the property will be cleared for the file. Subsequent calls to "Read & Get
+     *        Properties" will not return this property unless it is explicitly set on that file
+     *        again.
+     */
+    std::string ContentMD5;
+
+    /**
+     * @brief The lease ID must be specified if there is an active lease.
+     */
+    std::string LeaseId;
+
+    /**
+     * @brief Sets the blob's cache control. If specified, this property is stored with the
+     *        blob and returned with a read request.
+     */
+    std::string CacheControl;
+
+    /**
+     * @brief Sets the blob's content type. If specified, this property is stored
+     *        with the blob and returned with a read request.
+     */
+    std::string ContentType;
+
+    /**
+     * @brief Sets the blob's Content-Disposition header.
+     */
+    std::string ContentDisposition;
+
+    /**
+     * @brief Sets the blob's content encoding. If specified, this property is stored with
+     *        the blob and returned with a read request.
+     */
+    std::string ContentEncoding;
+
+    /**
+     * @brief Set the blob's content language. If specified, this property is stored with
+     *        the blob and returned with a read request.
+     */
+    std::string ContentLanguage;
```

Comment:
Maybe these 5 headers and `ContentMD5` should be put in a struct and shared among ADLS, Blob and File service?

Reply:
Yes, this is currently not a priority. Opened an issue to track: [Refinement] Aggregate shared http headers into one struct. #182

Reply:
I closed that issue. I think it's a bad idea to make DataLake depend on Blobs. Please define it by yourself.

Reply:
This would be a breaking change if don't do it now. So I think it's priority.

Reply:
Updated the issue, no longer a breaking change.